### ISSUE TITLE
Ruffle Error on https://armorgames.com/play/14170/arkandian-explorer?tag-referral=strategy

### ISSUE THREAD

**Issue by colares2** [ruffle-rs/ruffle#2939]
# Error Info
Error name: Error
Error message: panicked at 'index out of bounds: the len is 0 but the index is 4294967295', core/src/avm1/object/script_object.rs:816:17
Error stack:
```
J@https://agi.armorgames.com/ruffle/core.ruffle.9eec0e753f06015bcd55.js:1:20928
__wbg_new_df6e6ab7a65c4c4d@https://agi.armorgames.com/ruffle/ruffle.js:1:38584
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[1474]:0x278771
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[1491]:0x27b7e8
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[4122]:0x345dcc
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[3198]:0x3223f0
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[3748]:0x33c41e
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[4464]:0x34b42e
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[4339]:0x349acb
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[4466]:0x34b492
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[3914]:0x341192
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[1415]:0x26e5a8
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[514]:0x1591d9
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[449]:0x12fd53
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[449]:0x12fcd2
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[449]:0x12f761
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[449]:0x12f761
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[449]:0x12f761
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[347]:0xd1e63
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[346]:0xd0d1f
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[874]:0x1f0078
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[1438]:0x2727e7
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[1827]:0x2ac18b
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[280]:0x1d4bd
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[843]:0x1e6148
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[412]:0x112ce1
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[924]:0x1ff689
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[3225]:0x323bb8
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[740]:0x1c210d
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[4371]:0x34a1c1
@https://agi.armorgames.com/ruffle/cc0a08d53db75ad5d75f.module.wasm:wasm-function[4641]:0x34d22a
I@https://agi.armorgames.com/ruffle/core.ruffle.9eec0e753f06015bcd55.js:1:18043
f@https://agi.armorgames.com/ruffle/core.ruffle.9eec0e753f06015bcd55.js:1:17764

```

# Player Info
Allows script access: true

# Page Info
Page URL: https://armorgames.com/play/14170/arkandian-explorer?tag-referral=strategy
SWF URL: https://cache.armorgames.com/files/games/arkandian-explorer-14170.swf?v=1373587528

# Browser Info
Useragent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:85.0) Gecko/20100101 Firefox/85.0
OS: Win32

# Ruffle Info
Version: 0.1.0
Name: nightly 2021-01-06
Channel: nightly
Built: 2021-01-06T00:49:08.191Z
Commit: b84ed23106b764d1dad0ed682ad7e6789a95009a

**Comment by Poopooracoocoo** [comment]
When did you encounter this crash?

**Comment by colares2** [comment]
started yesterday, and still, the first time I played the game, it worked, but some hours later when I tried to play again this crash occurred, for both Firefox and Google Chrome.

**Comment by RomHac** [comment]
I got the same thing, it worked fine for a few hours, but glitchy squares started appearing on-screen (think missingno but colourful), and the frame rate dropped considerably. I wrote off the lag as something all games do and thought the glitchy effects were part of the game, but when I tried to reload my page in an effort to reduce the lag I got an error message.

**Comment by Poopooracoocoo** [comment]
I think this'll be fixed by #3099 :)

**Comment by Herschel** [comment]
Should be fixed by #3099.
